tensorboard/components/tf_backend/requestManager.ts

```diff
@@ -13,287 +13,302 @@ See the License for the specific language governing permissions and
-  public body?: any;
+    /** The request body to use. This is the object that is passed to the
+     * XMLHttpRequest.send() method. If not given the 'send' method is called
+     * without an argument.
+     */
+    public body?: any;
 
-  /** If specified, this will be the value set in the
-   * XMLHttpRequest.withCredentials property.
-   */
-  public withCredentials?: boolean;
+    /** If specified, this will be the value set in the
+     * XMLHttpRequest.withCredentials property.
+     */
+    public withCredentials?: boolean;
 
-  // Validates this object. Throws InvalidRequestOptionsError on error.
-  public validate() {
-    if (this.methodType === HttpMethodType.GET) {
-      // We don't allow a body for a GET.
-      if (this.body) {
-        throw new InvalidRequestOptionsError(
-          'body must be missing for a GET request.');
+    // Validates this object. Throws InvalidRequestOptionsError on error.
+    public validate() {
+      if (this.methodType === HttpMethodType.GET) {
+        // We don't allow a body for a GET.
+        if (this.body) {
+          throw new InvalidRequestOptionsError(
+            'body must be missing for a GET request.'
+          );
+        }
       }
+      // We allow body-less or contentType-less POSTs even if they don't
+      // make much sense.
     }
-    // We allow body-less or contentType-less POSTs even if they don't
-    // make much sense.
   }
-}
 
-export class RequestManager {
-  private _queue: ResolveReject[];
-  private _maxRetries: number;
-  private _nActiveRequests: number;
-  private _nSimultaneousRequests: number;
+  export class RequestManager {
+    private _queue: ResolveReject[];
+    private _maxRetries: number;
+    private _nActiveRequests: number;
+    private _nSimultaneousRequests: number;
 
-  constructor(nSimultaneousRequests = 10, maxRetries = 3) {
-    this._queue = [];
-    this._nActiveRequests = 0;
-    this._nSimultaneousRequests = nSimultaneousRequests;
-    this._maxRetries = maxRetries;
-  }
+    constructor(nSimultaneousRequests = 10, maxRetries = 3) {
+      this._queue = [];
+      this._nActiveRequests = 0;
+      this._nSimultaneousRequests = nSimultaneousRequests;
+      this._maxRetries = maxRetries;
+    }
 
-  /**
-   * Gives a promise that loads assets from given url (respects queuing). If
-   * postData is provided, this request will use POST, not GET. This is an
-   * object mapping POST keys to string values.
-   */
-  public request(url: string, postData?: {[key: string]: string}):
-      Promise<any> {
-    const requestOptions = requestOptionsFromPostData(postData);
-    return this.requestWithOptions(url, requestOptions);
-  }
+    /**
+     * Gives a promise that loads assets from given url (respects queuing). If
+     * postData is provided, this request will use POST, not GET. This is an
+     * object mapping POST keys to string values.
+     */
+    public request(
+      url: string,
+      postData?: {[key: string]: string}
```

Comment:
Is it impossible to get trailing comma here too?
Rationale is the same as other trailing commas: I want git-diff to be minimal when adding/removing arguments.

Reply:
I wish, but our version of TypeScript doesn’t support that:

```
ERROR: ${PWD}/tensorboard/plugins/histogram/tf_histogram_dashboard/BUILD:7:1: Compiling 34 TypeScript files //tensorboard/plugins/histogram/tf_histogram_dashboard:tf_histogram_dashboard failed (Exit 1) execrooter failed: error executing command bazel-out/host/bin/tensorboard/scripts/execrooter bazel-out/k8-fastbuild/bin/tensorboard/plugins/histogram/tf_histogram_dashboard/tf_histogram_dashboard-tsc-execroot.json ... (remaining 16 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
../../../../../../../../../../../../../../tmp/tmp_VP6Gz/tf-histogram-dashboard/histogramCore.ts(23,11): error TS1009: Trailing comma not allowed.
../../../../../../../../../../../../../../tmp/tmp_VP6Gz/tf-histogram-dashboard/histogramCore.ts(29,26): error TS1009: Trailing comma not allowed.
```

See:

  - <https://github.com/prettier/prettier/issues/1820>
  - <https://github.com/microsoft/TypeScript/issues/16152>

If at some point we upgrade our toolchains, we can flip this flag and
re-Prettify; the diff is comparatively small (1800 lines).

Reply:
Wow, didn't realize our version of tsc didn't support this. Okay, LGTM.